nilearn/regions/parcellations.py

```diff
@@ -366,6 +367,13 @@ def _raw_fit(self, data):
             self.connectivity_ = connectivity
         # Avoid 0 label
         labels = labels + 1
+
+        # Check that appropriate number of labels were created
+        if len(labels) != self.n_parcels:
+            n_parcels_warning = ('The number of generated labels does not '
```

Comment:
...labels do not match...

Reply:
I think "does" is correct, here. It's much more clear-cut if we drop the qualifier "of generated labels"
> The number ... does

Reply:
I don't have an opinion about the "of generated labels", you decide.
With plural objects (number of, labels), do is the correct form. Does is used when the subject is singular. "The generated label does not match"

Reply:
I don't want this to devolve into a grammar conversation :laughing: 

"The number" is a singular noun was my point. But I'm going to resolve this and we can revisit if it's unclear down the line !

Reply:
Number implies plurality. 
Okay,
We can do a grammar cage match next time we meet!